nncf/quantization/passes.py

```diff
@@ -64,26 +64,26 @@ def remove_shapeof_subgraphs(
         if node.metatype in shapeof_metatypes:
             shape_of_nodes.append(node)
             continue
-        if node in infer_nodes:
+        if node.node_name in infer_nodes:
```

Comment:
@KodiaqQ These changes are necessary because a node from quantized_model_graph contains NumPy arrays in its attributes. Therefore, it is not possible to perform the `NNCFNode.__eq__()` method due to this comparison
```python
self.attributes == other.attributes
```
I have not changed `NNCFNode.__eq__()` method because these attributes are custom (they were added by algorithm).